pkg/actuators/machine/reconciler.go

```diff
@@ -187,7 +187,30 @@ func (r *Reconciler) update() error {
 
 	klog.Infof("Updated machine %s", r.machine.Name)
 
-	r.machineScope.setProviderStatus(newestInstance, conditionSuccess())
+	vpc, err := r.awsClient.DescribeVpcs(&ec2.DescribeVpcsInput{
+		VpcIds: []*string{newestInstance.VpcId},
+	})
+	if err != nil {
+		klog.Errorf("%s: error describing vpc: %v", r.machine.Name, err)
+		return err
+	}
+
+	dhcp, err := r.awsClient.DescribeDHCPOptions(&ec2.DescribeDhcpOptionsInput{
+		DhcpOptionsIds: []*string{vpc.Vpcs[0].DhcpOptionsId},
+	})
+	if err != nil {
+		klog.Errorf("%s: error describing dhcp: %v", r.machine.Name, err)
+		return err
+	}
+	var domainName string
+	for _, i := range dhcp.DhcpOptions[0].DhcpConfigurations {
```

Comment:
How hard would it be to add a test that covers this within the reconciler tests?

Reply:
It would be fairly non trivial.  I would prefer to refactor update() from a method to a function in order to make testing here easier.  We're over using methods and it makes mocking tests much more burdensome.

So, I don't think the juice is worth the squeeze here.